src/Installer/Microsoft.Dotnet.Installation/Internal/DotnetArchiveExtractor.cs

```diff
@@ -56,68 +54,155 @@ public void Prepare()
         }
     }
     public void Commit()
-    {
-        Commit(GetExistingSdkVersions(_request.InstallRoot));
-    }
-
-    public void Commit(IEnumerable<ReleaseVersion> existingSdkVersions)
     {
         using var activity = InstallationActivitySource.ActivitySource.StartActivity("DotnetInstaller.Commit");
 
-
         using (var progressReporter = _progressTarget.CreateProgressReporter())
         {
             var installTask = progressReporter.AddTask($"Installing .NET SDK {_resolvedVersion}", maxValue: 100);
 
             // Extract archive directly to target directory with special handling for muxer
-            ExtractArchiveDirectlyToTarget(_archivePath!, _request.InstallRoot.Path!, existingSdkVersions, installTask);
+            ExtractArchiveDirectlyToTarget(_archivePath!, _request.InstallRoot.Path!, installTask);
             installTask.Value = installTask.MaxValue;
         }
     }
 
-    /**
-     * Extracts the archive directly to the target directory with special handling for muxer.
-     * Combines extraction and installation into a single operation.
-     */
-    private string? ExtractArchiveDirectlyToTarget(string archivePath, string targetDir, IEnumerable<ReleaseVersion> existingSdkVersions, IProgressTask? installTask)
+    /// <summary>
+    /// Extracts the archive directly to the target directory with special handling for muxer.
+    /// Combines extraction and installation into a single operation.
+    /// </summary>
+    private void ExtractArchiveDirectlyToTarget(string archivePath, string targetDir, IProgressTask? installTask)
     {
         Directory.CreateDirectory(targetDir);
 
-        var muxerConfig = ConfigureMuxerHandling(existingSdkVersions);
+        string muxerName = DotnetupUtilities.GetDotnetExeName();
+        string muxerTargetPath = Path.Combine(targetDir, muxerName);
+        string muxerTempPath = muxerTargetPath + ".tmp";
```

Comment:
instead of doing .tmp and then failing in the rare event we can no longer delete the .tmp file, we should generate or use a unique per session hash (note that we already have one of those!)

Reply:
Where is the unique per session hash?

For now I just added a new GUID to the filename.

Reply:
It used to be in InstallerOrchestratorSingleton but it's gone now - great question. This looks good